plugin/pkg/admission/taint/admission.go

```diff
@@ -0,0 +1,282 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+Copyright 2015 CoreOS, Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package taint
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/admission"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/rbac"
+	"k8s.io/kubernetes/pkg/apis/rbac/validation"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/rbac/unversioned"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	apierrors "k8s.io/kubernetes/pkg/api/errors"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+)
+
+const TaintKey string = "Untrusted"
+
+func newRbacClient(client clientset.Interface) validation.AuthorizationRuleResolver {
+	c := &rbacClient{client.Rbac()}
+	return validation.NewDefaultRuleResolver(c, c, c, c)
+}
+
+type rbacClient struct {
+	client unversioned.RbacInterface
+}
+
+func (c *rbacClient) GetClusterRole(ctx api.Context, id string) (*rbac.ClusterRole, error) {
+	return c.client.ClusterRoles().Get(id)
+}
+
+func (c *rbacClient) ListClusterRoleBindings(ctx api.Context, options *api.ListOptions) (*rbac.ClusterRoleBindingList, error) {
+	return c.client.ClusterRoleBindings().List(*options)
+}
+
+func (c *rbacClient) GetRole(ctx api.Context, id string) (*rbac.Role, error) {
+	ns, ok := api.NamespaceFrom(ctx)
+	if !ok {
+		return nil, errors.New("no namespace found")
+	}
+	role, err := c.client.Roles(ns).Get(id)
+	if err != nil {
+		return nil, fmt.Errorf("get role: %v", err)
+	}
+	return role, nil
+}
+
+func (c *rbacClient) ListRoleBindings(ctx api.Context, options *api.ListOptions) (*rbac.RoleBindingList, error) {
+	ns, ok := api.NamespaceFrom(ctx)
+	if !ok {
+		return nil, errors.New("no namespace found")
+	}
+	roleBindings, err := c.client.RoleBindings(ns).List(*options)
+	if err != nil {
+		return nil, fmt.Errorf("list role bindings: %v", err)
+	}
+	return roleBindings, nil
+}
+
+func init() {
+	admission.RegisterPlugin("TaintAdmit", func(client clientset.Interface, config io.Reader) (admission.Interface, error) {
+		return NewTaintAdmit(client, config), nil
+	})
+}
+
+// TaintAdmit is an implementation of admission.Interface which taints nodes at creation time
+type taintAdmit struct {
+	client clientset.Interface
+}
+
+// Determine whether a node is trusted
+func isTrusted(node *api.Node) (bool, error) {
```

Comment:
This is defined both in this package and in the tpm-manager. Can we combine these into a single location?